[Buttons] Implement a color themer for an MDCRaisedButton

yarneo opened the pull request:
Addition of a unit test, and updated example to use the new themer.

Pivotal: https://www.pivotaltracker.com/story/show/156640797

![simulator screen shot - iphone 8 - 2018-04-10 at 16 13 40](https://user-images.githubusercontent.com/4066863/38558964-6e3534ae-3cda-11e8-9a60-c2b89ce7889b.png)

yarneo committed:
added color themer for MDCButton

yarneo committed:
added color themer for MDCButton

yarneo committed:
added color themer for MDCButton

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update ButtonsColorThemerTests.m

yarneo committed:
Update MDCButtonColorThemer.h

yarneo committed:
Update MDCButtonColorThemer.h

yarneo committed:
Update MDCButtonColorThemer.h

Commit:
Automatic changelog preparation for release.

yarneo committed:
updated disabledAlpha

yarneo committed:
updated disabledAlpha

yarneo committed:
updated disabledAlpha

yarneo committed:
remove disabled alpha

yarneo committed:
remove disabled alpha

yarneo committed:
remove disabled alpha

yarneo committed:
raised button theming

yarneo committed:
raised button theming

yarneo committed:
raised button theming

yarneo committed:
raised button theming

Commit:
Update changelog.

Commit:
Bump the release.

yarneo committed:
indent

yarneo committed:
indent

yarneo committed:
indent

romoore commented on components/Buttons/src/ColorThemer/MDCButtonColorThemer.h:
```diff
@@ -30,6 +30,14 @@
                         toButton:(nonnull MDCButton *)button;
 
+/**
+ Applies a color scheme to theme to a MDCRaisedButton.
```
Super-nit: Should be "...an MDCRaisedButton."

romoore commented on components/Buttons/src/ColorThemer/MDCButtonColorThemer.h:
```diff
@@ -30,6 +30,14 @@
+/**
+ Applies a color scheme to theme to a MDCRaisedButton.
+
+ @param colorScheme The color scheme to apply to MDCRaisedButton.
```
LINT: Missing `@param` for `raisedButton`
Nit: This comment should be, "...apply to @c raisedButton."

romoore commented on components/Buttons/src/ColorThemer/MDCButtonColorThemer.m:
```diff
@@ -33,6 +33,11 @@ + (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
   button.disabledAlpha = 1.f;
 }
 
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
```
Why do we have a specific method for `toRaisedButton:` if there is no such class and it just calls the general themer?

romoore commented on components/Buttons/tests/unit/ButtonsColorThemerTests.m:
```diff
@@ -62,4 +62,41 @@ - (void)testMDCButtonColorThemer {
   XCTAssertEqual(button.disabledAlpha, 1.f);
 }
 
+- (void)testMDCRaisedButtonColorThemer {
```
Nit: The name should probably be something like `-testApplySemanticColorSchemeToRaisedButton`. There is no `MDCRaisedButtonColorThemer` class.

romoore commented on components/Buttons/tests/unit/ButtonsColorThemerTests.m:
```diff
@@ -62,4 +62,41 @@ - (void)testMDCButtonColorThemer {
+  [button setBackgroundColor:UIColor.redColor forState:UIControlStateHighlighted];
+  [button setBackgroundColor:UIColor.blueColor forState:UIControlStateSelected];
+  [button setBackgroundColor:UIColor.darkGrayColor forState:UIControlStateDisabled];
+
```
Please add a few more states and verify that they remain unaffected by the tests.  For example, `UIControlStateSelected | UIControlStateHighlighted`

romoore commented on components/Buttons/tests/unit/ButtonsColorThemerTests.m:
```diff
@@ -62,4 +62,41 @@ - (void)testMDCButtonColorThemer {
+  XCTAssertEqual([button titleColorForState:UIControlStateNormal], colorScheme.onPrimaryColor);
+  XCTAssertEqual([button titleColorForState:UIControlStateHighlighted], colorScheme.onPrimaryColor);
+  XCTAssertEqual([button titleColorForState:UIControlStateSelected], colorScheme.onPrimaryColor);
+  XCTAssert(
```
I believe XCTAssertEqualObjects works with colors just fine as long as the alpha is the same precision (single/double) in both initializer invocations.

jverkoey commented on components/Buttons/src/ColorThemer/MDCButtonColorThemer.m:
```diff
@@ -33,6 +33,11 @@ + (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
 }
 
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                        toRaisedButton:(nonnull MDCRaisedButton *)raisedButton {
```
minor nit: indentation

romoore commented on components/Buttons/tests/unit/ButtonsColorThemerTests.m:
```diff
@@ -62,4 +62,41 @@ - (void)testMDCButtonColorThemer {
+  XCTAssert(
+      CGColorEqualToColor([button backgroundColorForState:UIControlStateDisabled].CGColor,
+                          [colorScheme.onSurfaceColor colorWithAlphaComponent:0.12f].CGColor));
+  XCTAssertEqual(button.disabledAlpha, 1.f);
```
Please use XCTAssertEqualWithAccuracy for floating point comparison.

mohammadcazig committed:
[NavigationBar] typography implementation. (#3304)

* Navigation Bar Typography Themer implementation.
Tests and example added.

* fixing merge conflict mistake and tests.

jverkoey committed:
[kokoro] Validate the CocoaPods podspec (#3301)

We will now validate the podspec when building against Xcode 9.2.0.

This change also adds support for running individual Xcode instances against CocoaPods.

Commit:
Merge branch 'release-candidate' into develop

jverkoey committed:
[travis] Use verbose output when linting our podspec. (#3314)

Travis doesn't like when commands don't emit any output for more than 10 minutes. Our pod lib lint step takes 15-20 minutes. Enabling verbose output will allow travis to know that the lint step is actually doing something.

mohammadcazig committed:
BottomNavigationBar Typography Implementation, (#3311)

Tests and example added.

jverkoey committed:
[Catalog] Use new MDCColorScheming APIs for theming the catalog (#3235)

Updates the AppTheme to make use of the MDCColorScheming APIs.

Pivotal story: https://www.pivotaltracker.com/story/show/156522700

![simulator screen shot - iphone 8 plus - 2018-04-10 at 13 17 38](https://user-images.githubusercontent.com/45670/38572619-9364b66e-3cc1-11e8-877c-d8ef81cc59a3.png)

![simulator screen shot - iphone 8 plus - 2018-04-10 at 13 17 47](https://user-images.githubusercontent.com/45670/38572624-94f0411a-3cc1-11e8-83ac-78497fdd80f1.png)

jmdetloff committed:
Ensure the mdc icon fades out before crossing status bar (#3257)

jmdetloff committed:
Fix Flexible Header Configurator catalog example (#3254)

Currently this example, when configured such that:

Basic Behavior
Can over-extend - ON
In front of infinite content - OFF
Hide status bar - OFF

Shift Behavior
Enabled - ON
Enabled with status bar - OFF
Header content is important - OFF

the header text overlaps the status bar. This change ensures the minimum height is always great enough to prevent that.

mohammadcazig committed:
[TextFields] Typography Themer implementation. (#3303)

* TextFields Typography Themer added.
Tests and example included.

* Removing extra checks, comment the reason behind checking pointsize

* fixing typo.

mohammadcazig committed:
Chips typography implementation. (#3312)

Tests and example added.

yarneo committed:
[Snackbar] Implement a semantic color scheme color themer API. (#3287)

Addition of a unit test, and updated the examples to use the new themer. Removed older stub color themer that was a no-op.

Pivotal: https://www.pivotaltracker.com/story/show/156171666

Commit:
Automatic changelog preparation for release.

jverkoey committed:
[kokoro] Add a cocoapods linter job target. (#3317)

This will allow us to lint the podspec independently of our build jobs.

yarneo committed:
[Buttons] Implement a typography themer. (#3318)

Addition of a unit test, and updated the examples to use the new typography themer.

Pivotal: https://www.pivotaltracker.com/story/show/156171586

Commit:
Update changelog.

Commit:
Bump the release to 54.3.0.

Commit:
Merge branch 'release-candidate' into develop

yarneo committed:
[Buttons] Implement a color themer for an MDCFlatButton (#3308)

Addition of a unit test, and updated example to use the new themer.

Pivotal: https://www.pivotaltracker.com/story/show/156640819

jverkoey committed:
[travis] Remove pod lib lint step. (#3323)

It causes Travis to fail because the logs get too long. If we disable logs, travis fails because the invocation takes too long. Soo...we can't lint in travis.

ianegordon commented:
Please rebase / fix merge conflicts when addressing @romoore comments.

Commit:
[ThumbTrack] Fix valueLabel backgroundColor (#3322)

The discrete value label's background color was not being set when
setting the `primaryColor`. Since there is currently no public API (via
MDCSlider) to set the color, this can cause undesirable color mismatches
in clients.

**Before**
![slider-valuelabel-bug](https://user-images.githubusercontent.com/1753199/38622949-66f54cd6-3d72-11e8-8c33-df8b28ddb319.png)

**After**
![slider-valuelabel-fixed](https://user-images.githubusercontent.com/1753199/38622957-6c0c4e2c-3d72-11e8-9e9c-34c6a3b14dbc.png)


Fixes #3321

yarneo committed:
update

yarneo committed:
update

yarneo commented:
some rebase issues, will reopen once ready.
